checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -223,6 +253,10 @@ protected void refineGT(
             knownToBeLessThanLengthOf(array, right, store, in);
         }
 
+        if (right instanceof NumericalAdditionNode || right instanceof NumericalSubtractionNode) {
+            propagateToSubterms((BinaryOperationNode) right, refinedRight, store);
```

Comment:
Instead of creating a new method, I think the `knownToBeLessThanLengthOf` methods could be generalized to take a qualifier.  (And perhaps renamed to `propagateToSubterms` or maybe `propagateToOperands`).